cmd/x-check-mailserver/work.go

```diff
@@ -40,163 +43,105 @@ func NewWorkUnit(mailEnode string, config *params.NodeConfig) *WorkUnit {
+		return err
 	}
 
-	// TODO(adam): change to regular fanout. It might happen that a signal
-	// is received before the filter is setup.
-	signals, cancelSignalsFilter := mailSignals.Filter([]byte(reqID))
-	defer cancelSignalsFilter()
-
-	start := time.Now()
-
-	var lastEnvelopeID []byte
-
-	for {
-		select {
-		case m := <-messages:
-			log.Debug("received a message", "hash", hex.EncodeToString(m.Hash))
-			u.Messages = append(u.Messages, m)
-		case <-time.After(time.Duration(reqTimeout) * time.Second):
-			// As we can not predict when messages finish to come in,
-			// we timeout after some time.
-			// If lastEnvelopeID is found amoung received messages,
-			// it's a successful request. Otherwise, an error is returned.
-			for i, m := range u.Messages {
-				if bytes.Equal(lastEnvelopeID, m.Hash) {
-					log.Info("received a message equal to lastEnvelopeID",
-						"hash", hex.EncodeToString(lastEnvelopeID),
-						"index", i,
-						"messagesCount", len(u.Messages))
-					return u.stopNode()
-				}
-			}
-			return fmt.Errorf("did not receive a message equal to lastEnvelopeID")
-		case err := <-merge(messageSubErrs...):
-			return fmt.Errorf("subscription for messages errored: %v", err)
-		case s := <-signals:
-			switch s.Type {
-			case signal.EventMailServerRequestCompleted:
-				lastEnvelopeID = s.LastEnvelopeID
-
-				log.Info("received EventMailServerRequestCompleted", "latency", time.Since(start), "enode", u.MailServerEnode, "lastEnvelopeID", lastEnvelopeID)
-
-				if allZeros(lastEnvelopeID) {
-					log.Info("lastEnvelopeID is empty so return early")
-					return u.stopNode()
-				}
-			case signal.EventMailServerRequestExpired:
-				return fmt.Errorf("request for messages expired")
-			}
+	for _, m := range resp.Messages {
+		u.MessageHashes = append(u.MessageHashes, types.FromHex(m.ID))
+	}
+
+	for _, rawResp := range resp.RawMessages {
+		for _, m := range rawResp.Messages {
+			u.MessageHashes = append(u.MessageHashes, m.ID)
 		}
 	}
+
+	return nil
 }
 
 func (u *WorkUnit) startNode() error {
 	u.node = node.New()
-	if err := u.node.Start(u.config); err != nil {
+	if err := u.node.Start(u.config, &accounts.Manager{}); err != nil {
 		return fmt.Errorf("failed to start a node: %v", err)
 	}
+	return nil
+}
 
-	rpcClient, err := u.node.GethNode().Attach()
+func (u *WorkUnit) stopNode() error {
+	return u.node.Stop()
+}
+
+func (u *WorkUnit) startMessenger() error {
+	key, err := crypto.GenerateKey()
 	if err != nil {
-		return fmt.Errorf("failed to get an rpc: %v", err)
+		return err
 	}
-	u.shh = shhclient.NewClient(rpcClient)
-
-	shhextService, err := u.node.ShhExtService()
+	u.key = key
+	node := gethbridge.NewNodeBridge(u.node.GethNode())
+	db, err := sqlite.OpenInMemory()
 	if err != nil {
-		return fmt.Errorf("failed go get an shhext service: %v", err)
+		return err
 	}
-	u.shhextAPI = shhext.NewPublicAPI(shhextService)
-
+	messenger, err := protocol.NewMessenger(key, node, "instalation-01", protocol.WithDatabase(db))
```

Comment:
Could we lower the noise by getting rid of theshitloads of `DEBUG` messages.
```suggestion
	// Create a custom logger to suppress DEBUG messages
	logger, _ := zap.NewProduction()

	options := []protocol.Option{
		protocol.WithDatabase(db),
		protocol.WithCustomLogger(logger),
	}

	messenger, err := protocol.NewMessenger(key, node, "instalation-01", options...)
```
Why even is the default level `DEBUG`?

Also, is it sensible to leave the installation ID hardcoded?

Reply:
`installation-id` is not used but I need to pass something.